Dependencies/xr/Source/ARKit/XR.mm

```diff
@@ -622,12 +624,37 @@ fragment float4 fragmentShader(RasterizerData in [[stage_in]],
         }
     }
 
+    struct XrContextARKit : public IXrContextARKit {
+        bool Initialized{true};
+        ARSession* Session{nullptr};
+        ARFrame* Frame{nullptr};
+
+        bool IsInitialized() const override
+        {
+            return Initialized;
+        }
+
+        ARSession* XrSession() const override
+        {
+            return Session;
+        }
+
+        ARFrame* XrFrame() const override
+        {
+            return Frame;
+        }
+
+        virtual ~XrContextARKit() = default;
+    };
+
     struct System::Impl {
     public:
+        std::shared_ptr<XrContextARKit> XrContext{std::make_shared<XrContextARKit>()};
```

Comment:
Is this ownership actually conceptually shared? If not, could this be a simpler type?

Reply:
I don't think it will be possible to hand a shared pointer across dll/lib boundaries. That going said, from a legibility standpoint id prefer this to stay a pointer compared to a memory reference to a declared struct. We could change this to a unique_ptr if that's preferred over shared_ptr?

Reply:
talked offline, opted to switch to unique_ptr. this should now be fixed.